app/upload/video/camera.tsx

```diff
@@ -0,0 +1,191 @@
+import { CameraView, CameraType, useCameraPermissions, FlashMode, useMicrophonePermissions } from 'expo-camera';
+import React, { useState, useRef, useEffect } from 'react';
+import { Text, View, Pressable, Platform} from 'react-native';
+import { supabase } from '../../../lib/supabase';
+import { SafeAreaView } from 'react-native-safe-area-context';
+import { router } from 'expo-router';
+import { GestureDetector, Gesture } from 'react-native-gesture-handler';
+import { useMediaLibraryPermissions, launchImageLibraryAsync, MediaTypeOptions } from 'expo-image-picker';
+import { decode } from 'base64-arraybuffer';
+import { readAsStringAsync } from 'expo-file-system';
+import TopRow from '../../../components/upload/TopRow';
+import BottomRow from '../../../components/upload/video/BottomRow';
+import Camera from '../../../components/upload/video/Camera';
+
+export default function Page() {
+    const cameraRef = useRef<CameraView>(null);
+    const [cameraPermission, requestCameraPermission] = useCameraPermissions();
+    const [mediaLibraryPermission, requestMediaLibraryPermission] = useMediaLibraryPermissions();
+    const [useMicrophonePermission, requestMicrophonePermission] = useMicrophonePermissions();
+    const [isRecording, setIsRecording] = useState(false);
+    const [videoUri, setVideoUri] = useState<string | null>(null);
+    const [facing, setFacing] = useState<CameraType>('back');
+    const [flash, setFlash] = useState<FlashMode>('off');
+    const [zoom, setZoom] = useState<number>(0);
+    const [lastZoom, setLastZoom] = useState<number>(0);
+    const [timer, setTimer] = useState<number>(60)
+    const timerRef = useRef<NodeJS.Timeout | null>(null);
+    const currentDate = new Date();
+    const currentDateIso = `${currentDate.getFullYear()}-${currentDate.getMonth() + 1}-${currentDate.getDate()}T00:00:00`;
+
+    useEffect(() => {
+        if (isRecording) {
+            timerRef.current = setInterval(() => {
+                setTimer(prev => {
+                    if (prev <= 1) {
+                        clearInterval(timerRef.current!);
+                        setIsRecording(false);
+                        return 60;
+                    }
+                    return prev - 1;
+                });
+            }, 1000);
+        } else {
+            if (timerRef.current) {
+                clearInterval(timerRef.current);
+                setTimer(60);
+            }
+        }
+
+        return () => {
+            if (timerRef.current) {
+                clearInterval(timerRef.current);
+            }
+        };
+    }, [isRecording]);
+
+
+    if (!cameraPermission || !mediaLibraryPermission || !useMicrophonePermission) {
+        //Change to loading state
+        return <View />
+    }
+
+    async function requestPermission() {
```

Comment:
Consider abstracting the permissions request into a helper function, which will improve readability and make it easier to manage permissions as a group like:
async function requestAllPermissions() {
    await Promise.all([
        requestCameraPermission(),
        requestMediaLibraryPermission(),
        requestMicrophonePermission(),
    ]);
}